File: pinyougou-user-web/src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../user/findAll.do');		
	};
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../user/findPage.do?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findOne=function(id){
		return $http.get('../user/findOne.do?id='+id);
	};
	//增加 
	this.add=function(entity,smscode){
		return  $http.post('../user/add.do?smscode='+smscode,entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../user/update.do',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../user/delete.do?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../user/search.do?page='+page+"&rows="+rows, searchEntity);
	};



	//发送验证码
	this.sendCode=function(phone,code){
		return $http.get('../user/sendCode.do?phone='+phone);
	}


	/***
	* @Description: 修改用户的登陆密码
	* @Param:
	* @return:
	* @Author: WangRui
	* @Date: 2019/7/25
	*/

    this.updatePassword=function(entity){
        return  $http.post('../user/updatePassword.do',entity );
    }



    this.findUser=function(){
        return  $http.post('../user/findUser.do');
    }


    this.checkCode=function (code,phone) {
        return $http.get('../user/checkCode.do?code='+code+'&phone='+phone);
    }

    //查询所有省市区
    this.findPrv=function () {
        return $http.get('../user/findProvince.do');
    }
	//查询所有城市
    this.findCity=function (provinceId) {
        return $http.get('../user/findCity.do?provinceId='+provinceId);
    }
	//查询所有区
    this.findArea=function (cityId) {
        return $http.get('../user/findArea.do?cityId='+cityId);
    }

	
});

File: pinyougou-user-interface/src/main/java/com/pinyougou/user/service/UserService.java
package com.pinyougou.user.service;
import java.util.List;

import com.pinyougou.pojo.TbAreas;
import com.pinyougou.pojo.TbCities;
import com.pinyougou.pojo.TbProvinces;
import com.pinyougou.pojo.TbUser;

import entity.PageResult;
import entity.Result;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface UserService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbUser> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbUser user);
	
	
	/**
	 * 修改
	 */
	public void update(TbUser user);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbUser findOne(Long id);

	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbUser user, int pageNum,int pageSize);
	
	
	/**
	 * 发送短信验证码
	 * @param phone
	 */
	public void createSmsCode(String phone);
	
	/**
	 * 校验验证码
	 * @param phone
	 * @param code
	 * @return
	 */
	public boolean checkSmsCode(String phone,String code);



 
	/***
	* @Description: 修改密码
	* @Param: [user, username]
	* @return: void
	* @Author: WangRui
	* @Date: 2019/7/25
	*/ 
	public Result updatePassword(TbUser user,String username);
	
	
	/***
	* @Description: 修改手机的号码
	* @Param: [username, phone]
	* @return: void
	* @Author: WangRui
	* @Date: 2019/7/25
	*/
	public void updatePhone(String username,String phone);

	
	
    /***
    * @Description: 登陆的用户
    * @Param: [name]
    * @return: com.pinyougou.pojo.TbUser
    * @Author: WangRui
    * @Date: 2019/7/25
    */

	TbUser findUser(String name);

	/**
	 * 查询所有省份
	 * @return
	 */
	public List<TbProvinces> findProvince();

	/**
	 * 查询所有市
	 * @return
	 */
	public List<TbCities> findCity(String pid);

	/**
	 * 查询所有区县
	 * @return
	 */
	public List<TbAreas> findArea(String cid);

	/**
	 *
	 * @param orderId
	 * @return
	 */

	String getEmailFromOrderId(String orderId);
}
